README.md

```diff
@@ -421,7 +421,65 @@ By default, this library doesn't print anything to stdout/stderr, however user c
+
+Follow to next steps for configure on Apple side: 
+
+1. At first, you need login in to [developer account](https://developer.apple.com/account) and enable capabilities `Sign in with Apple` for specific App ID (in [Certificates, Identifiers & Profiles](https://developer.apple.com/account/resources/identifiers/list) section ).  Also, you need remember `App ID Prefix` (is a `TeamID` value) it's required for configure Apple auth provider.
+
+2. Create [Service ID](https://developer.apple.com/account/resources/identifiers/list/serviceId) and bind App ID which has enabled capabilities `Sign in with Apple`. Description field value will be display for end users when they are doing sign in. `Identifier` of  created service is a `Client ID`.
+
+3. Configure `Sign in with Apple` for created `Service ID`. Add domain, which use for auth with Apple provider. Domain must available only by `https`, on the standard port `443` only (other ports not accepted), has trusted certificate (self-signed certificate not allowed) and support TLS 1.2 or higher. Also you need add redirect URLs. It's must be registered with the `https` protocol only and trusted certificate.
+
+4. Register a [New Key](https://developer.apple.com/account/resources/authkeys/list) (private key) for `Sign in with Apple` feature and download it. After create a download a key impossible. Remember `Key ID`. This  key use for create [JWT](https://developer.apple.com/documentation/sign_in_with_apple/generate_and_validate_tokens#3262048) `Client Secret`
+
+5. You should make sure, that SPF (DNS record) for a domain (which used for `Sign in with Apple` and bind to `Service ID`) is verified. It's can be done it in `Certificates, Identifiers & Profiles` >> [`More`](https://developer.apple.com/account/resources/services/configure) section.
+
+After complete previous steps you can go to configure `Apple auth provider`. For it require parameters which must set in `AppleConfig`:
+
+* _ClientID_ (**required**) - Service ID identifier which use for `Sign with Apple`
+* _TeamID_ (**required**) - Identifier a developer account (use as prefix for all App ID)
+* _KeyID_ (**required**) - Identifier a generated key for `Sign with Apple`
+* _Scopes_ (optional) - The [scope](https://developer.apple.com/documentation/sign_in_with_apple/clientconfigi/3230955-scope) is the amount of user information requested from Apple. Right now, there are only two options name and email. You can request the user’s name or email. You can also choose to request both or neither.
+* _ClientSecretExpire_ (optional) - time in seconds when client secret will expire (default: 1800 secs = 30 minutes)
+
+```go 
+    // apple config parameters  
+	appleCfg := provider.AppleConfig{
```

Comment:
If some keys are required, your code must explicitly fail when they are not provided.

Reply:
Nvm, I see that it's doing the check already.